### PR TITLE
docs(contrib): add security.md

### LINKED ISSUE #2402: Create security issue reporting process and add SECURITY.md file
Need:
- A mailing list
- A subset of project maintainers to serve as security team
- A document with guidelines

See example from Helm project [here](https://github.com/helm/community/blob/master/SECURITY.md)

### PR DESCRIPTION
**Pull request by schristoff**
Signed-off-by: Sarah Christoff <sarah.christoff13@gmail.com>

**Description**:
Fixes #2402 

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
    yes

2. Is this a breaking change?
no

4. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?
n/a

### CODE REVIEW

**Comment by jaellio** [comment]
@schristoff could you please add a commit message or PR description to this PR? Thanks!

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+A maintainer will acknowledge your email within 48 hours, and will
+send a more detailed response within 72 hours indicating the next steps in 
+handling your report. After the initial reply to your report, the team
+will endeavor to keep you informed of the progress towards a fix and
+full announcement, and may ask for additional information or guidance.
+
+Report security vulnerabilities in third-party modules to the person or 
+team maintaining the module.
+
```
[the remark this answers is not in the document]
Comment (schristoff) [reply SECURITY.md:28]: This is super beneficial. I love all of these ideas. :)

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Open Service Mesh Security Policies and Procedures
+
+This document outlines security procedures and general policies for the
+Open Service Mesh Open Source projects as found on https://github.com/openservicemesh/osm.
```
Comment (allenlsy) [inline SECURITY.md:4]: nit: `open source` does not need to be capitalized.

File: SECURITY.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (schristoff) [reply SECURITY.md:38]: Updated with your suggestions! :)

**Comment by nojnhuh** [comment]
@phillipgibson Would you mind taking a look at this?

**Comment by phillipgibson** [comment]
A good start. I would also take a look at [Helm's](https://github.com/helm/community/blob/main/SECURITY.md) guide as well. I think we should have the sections on when and when not to send a report so there's clarity.

**Comment by schristoff** [comment]
@phillipgibson  let me know if these new sections are what you were thinking

**Review by phillipgibson — APPROVED** [review]

**Comment by shashankram** [comment]
@trstringer could you review this?

**Pull request merged**